Formula/easy-rsa.rb

```diff
@@ -0,0 +1,54 @@
+class EasyRsa < Formula
+  desc "CLI utility to build and manage a PKI CA"
+  homepage "https://github.com/OpenVPN/easy-rsa"
+  url "https://github.com/OpenVPN/easy-rsa/archive/v3.0.8.tar.gz"
+  sha256 "fd6b67d867c3b8afd53efa2ca015477f6658a02323e1799432083472ac0dd200"
+  license "GPL-2.0-only"
+  head "https://github.com/OpenVPN/easy-rsa.git"
+
+  depends_on "openssl@1.1"
+
+  def install
+    # Rename and edit vars.example file to define easy-rsa variables for homebrew
+    mv "easyrsa3/vars.example", "easyrsa3/vars"
+    inreplace "easyrsa3/vars" do |s|
+      s.gsub! "#set_var EASYRSA\t\"${0%/*}\"",
+              "set_var EASYRSA\t\"#{etc}/pki\""
+      s.gsub! "#set_var EASYRSA_OPENSSL\t\"openssl\"",
+              "set_var EASYRSA_OPENSSL\t\"#{Formula["openssl@1.1"].bin}/openssl\""
+      s.gsub! "#set_var EASYRSA_PKI\t\t\"$PWD/pki\"",
+              "set_var EASYRSA_PKI\t\"$EASYRSA\""
+    end
+
+    bin.install "easyrsa3/easyrsa"
+
+    (etc/"pki").install %w[
+      easyrsa3/openssl-easyrsa.cnf
+      easyrsa3/vars
+      easyrsa3/x509-types
+    ]
+
+    doc.install %w[
+      ChangeLog
+      COPYING.md
+      KNOWN_ISSUES
+      README.md
+      README.quickstart.md
+    ]
+
+    doc.install Dir["doc/*"]
+  end
+
+  def caveats
+    <<~EOS
+      Add the following to your ~/.bash_profile or ~/.zshrc file:
+        export EASYRSA=#{etc}/pki
+    EOS
+  end
+
+  test do
+    # Create a stub PKI directory in the test sandbox.
+    ENV["EASYRSA_PKI"] = testpath/"pki"
+    system bin/"easyrsa", "init-pki"
```

Comment:
Please check that something actually happens here.

Reply:
Will do. Also, I noticed that there are a couple of test scripts included, so I'll look into using one or both of those instead.

Reply:
We don't want to run upstream test scripts though, we want to check if something a user would do themselves is working.

Reply:
Okay, I think I need a bit of help on this one. If I test it manually, things seem to work:

```
$ rm -rf ~/pki
$ brew install easy-rsa
==> Downloading https://github.com/OpenVPN/easy-rsa/archive/v3.0.8.tar.gz
Already downloaded: /Users/chris/Library/Caches/Homebrew/downloads/9721939a50a850a6861cbb7f16953662d77aa181623cd96ebf85d5073ebf2d0e--easy-rsa-3.0.8.tar.gz
==> Caveats
Add the following to your ~/.bash_profile or ~/.zshrc file:
  export EASYRSA=/usr/local/opt/easy-rsa/share
==> Summary
🍺  /usr/local/Cellar/easy-rsa/3.0.8: 24 files, 134.7KB, built in 4 seconds
$ EASYRSA_PKI=~/pki easyrsa init-pki

Note: using Easy-RSA configuration from: /usr/local/opt/easy-rsa/share/vars

init-pki complete; you may now create a CA or requests.
Your newly created PKI dir is: /Users/chris/pki


$ ls -la ~/pki
total 32
drwx------    6 chris  staff   192 Nov  6 09:59 .
drwxr-xr-x+ 125 chris  staff  4000 Nov  6 09:59 ..
-rw-------    1 chris  staff  4616 Nov  6 09:59 openssl-easyrsa.cnf
drwx------    2 chris  staff    64 Nov  6 09:59 private
drwx------    2 chris  staff    64 Nov  6 09:59 reqs
-rw-------    1 chris  staff  4550 Nov  6 09:59 safessl-easyrsa.cnf
```

However, if I try to modify the test block to test for the presence of `openssl-easyrsa.cnf`:

```ruby
  test do
    # Create a stub PKI directory in the test sandbox.
    ENV["EASYRSA_PKI"] = testpath/"pki"
    system bin/"easyrsa", "init-pki"
    assert_predicate testpath/"pki/openssl-easyrsa.cnf", :exist?
  end
```
this is the result:
```
$ brew test easy-rsa
==> Testing easy-rsa
==> /usr/local/Cellar/easy-rsa/3.0.8/bin/easyrsa init-pki
Error: easy-rsa: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: <#<Pathname:/private/tmp/easy-rsa-test-20201106-75072-1w7b230/pki/openssl-easyrsa.cnf>>.exist? is true value expected but was
<false>
...
```
So obviously I don't understand something about how the Homebrew test environment works, or else it's failing in the test environment when the actual install is working. Thoughts?

Reply:
Okay, so this was failing because the test environment doesn't have access to the `#{shared}` directory where the config files are located (or at least the paths don't match anyway, because `#{shared}` is evaluated and then hard-coded into `vars` during the install), and thus couldn't locate the `openssl-easyrsa.cnf` to actually copy to the `pki/` directory during the test (although the rest of that directory structure was being created). 

It could be overcome by setting additional environment variables to define these things specifically for the test, but that kind of defeats the purpose, since they would all be different in the installed instance. The change I made is to simply parse the output of `easyrsa init-pki` for the success message `init-pki complete` which at least indicates that the executable is installed and runnable and not throwing errors.